R/modify-model-field.R

```diff
@@ -31,8 +31,9 @@
 #' @param .remove If `TRUE`, `.value` with be removed from the `.field` instead of added. `FALSE` by default. Cannot have both `.append` and `.remove` be true in the same call.
 #' @param .unique If `TRUE`, the default, de-duplicate `.mod[[.field]]` after adding new values. If `FALSE` duplicate values will be kept.
 #' @param .char_value If `TRUE`, check that `.value` (after unlisting) is a character vector.
+#' @param .bool_value If `TRUE`, check that `.value` (after unlisting) is a boolean vector.
 #' @export
-modify_model_field <- function(.mod, .field, .value, .append = TRUE, .remove = FALSE, .unique = TRUE, .char_value = TRUE) {
+modify_model_field <- function(.mod, .field, .value, .append = TRUE, .remove = FALSE, .unique = TRUE, .char_value = TRUE, .bool_value = FALSE) {
```

Comment:
Hmm, that's getting a bit unwieldy, and in my view it's not ideal having dependent options like `.bool_value` and `.char_value`, rather than rolling them into one.  On the other hand, this is exported, so we'd have to change this in a backward-compatible way.  That'd be a multi-step process to do that, and that doesn't seem worth it given that this function has more user-friendly wrappers on top.

So I think your choice of adding a new parameter is the way to go.

Reply:
If we wanted to add new data types to the model fields, would we have to do this in the future as well? For instance, would we have to include a `.numeric_value` if we wanted to report additional model coefficients?

Reply:
Right.  I don't know how likely it is that the need for another type would come up, but, if it did, the current approach would mean adding a new argument.  

The main other option is consolidating all the type-related arguments into a single one, but then we have to either 1) worry about adding a temporary compatibility layer, announcing the deprecation, and then removing the old args in a later release or 2) decide we're not worried about compatibility here.

Given those options, I'd say punt on consolidating it into a single argument.